connexion/security/security_handler_factory.py

```diff
@@ -362,18 +362,25 @@ def wrapper(request, token, required_scopes):
         return wrapper
 
     @classmethod
-    def verify_security(cls, auth_funcs, required_scopes, function):
+    def verify_security(cls, auth_funcs, function):
         @functools.wraps(function)
         def wrapper(request):
             token_info = cls.no_value
+            problem = None
             for func in auth_funcs:
-                token_info = func(request, required_scopes)
-                if token_info is not cls.no_value:
-                    break
+                try:
+                    token_info = func(request)
+                    if token_info is not cls.no_value:
+                        break
+                except (OAuthProblem, OAuthResponseProblem, OAuthScopeProblem) as err:
+                    problem = err
```

Comment:
This can be overwritten in subsequent iterations of the for loop, only retaining the last error, no? Is this a problem?

Reply:
Yes, you are correct: we can't return _all_ errors but wasn't sure what the best approach was here and whether it's enough to simply return _any_ error that was raised to at least return some info.

Some options:
- any/last specific error that was raised (current implementation)
- a generic `401`
- retain a `403` over a `401` as it's more specific (valid credentials but not enough privileges vs invalid/no credentials)

Reply:
I'm also not yet sure which errors to catch: if a security view function raises a custom defined error, this isn't caught here and another passing security scheme is possibly never reached. Perhaps it's better to simply catch any error?

Reply:
It's a bit hard to wrap my head around this since I'm not very familiar with the security implementation of Connexion.

But if I understand correctly, we loop through `auth_funcs` here and catch the errors since we want to apply them in an `or` fashion?

Not sure if we want to catch any error here. It would make the security check more robust as one broken function would not limit the use of the others. On the other hand, it can hide issues longer.

If we do catch any error, we should probably prioritize which one to return. E.g. if you have two functions which would return a `403` and `500` status code, we should return the `403` status code, since it's probably more relevant to the security method you're trying to use.